### PR TITLE
fix(payment): Fix Adyen invoice payment

### PR DESCRIPTION
**Pull request by ivannovosad**
## Context

Fix adyen payment failing jobs

### CODE REVIEW

File: lib/lago/adyen/error_handlable.rb [hunk 1]
```diff
@@ -4,13 +4,15 @@ module Lago
...
 
         code = res.response['errorType']
         message = res.response['message']
 
-        deliver_error_webhook(::Adyen::AdyenError.new(nil, nil, message, code))
-        result.service_failure!(code:, message:)
+        error = ::Adyen::AdyenError.new(nil, nil, message, code)
+        deliver_error_webhook(error)
+
+        false
```
Comment (vincent-pochet) [inline lib/lago/adyen/error_handlable.rb:15]: We are loosing the ability to track the initial error message from the service. Is it intended?

File: app/services/invoices/payments/adyen_service.rb [hunk 2]
```diff
@@ -77,8 +75,8 @@ def generate_payment_url
         return result unless should_process_payment?
 
         res = client.checkout.payment_links_api.payment_links(Lago::Adyen::Params.new(payment_url_params).to_h)
-        handle_adyen_response(res)
-
+        adyen_response = handle_adyen_response(res)
+        result.single_validation_failure!(error_code: 'payment_provider_error') unless adyen_response
```
Comment (vincent-pochet) [inline app/services/invoices/payments/adyen_service.rb:79]: Should it be a service_failure rather a validation failure?

File: app/services/payment_provider_customers/adyen_service.rb [hunk 3]
```diff
@@ -30,8 +30,8 @@ def generate_checkout_url(send_webhook: true)
       return result.not_found_failure!(resource: 'adyen_payment_provider') unless adyen_payment_provider
 
       res = client.checkout.payment_links_api.payment_links(Lago::Adyen::Params.new(payment_link_params).to_h)
-      handle_adyen_response(res)
-
+      adyen_response = handle_adyen_response(res)
+      result.single_validation_failure!(error_code: 'payment_provider_error') unless adyen_response
```
Comment (vincent-pochet) [inline app/services/payment_provider_customers/adyen_service.rb:34]: Same question regarding the error type

**Review by vincent-pochet — APPROVED** [review]
Could you provide a bit of context in the PR description? It could help for the review.

Small questions on my side, but looks good otherwise! :)

File: lib/lago/adyen/error_handlable.rb (hunk 1 above)
Comment (ivannovosad) [reply lib/lago/adyen/error_handlable.rb:15]: I added the error tracking.

**Review by vincent-pochet — APPROVED** [review]

File: app/services/invoices/payments/adyen_service.rb [hunk 4]
```diff
@@ -30,8 +30,8 @@ def create
         res = create_adyen_payment
         return result unless res
 
-        handle_adyen_response(res)
-        return result unless result.success?
+        adyen_success, adyen_error = handle_adyen_response(res)
```
Comment (jdenquin) [inline app/services/invoices/payments/adyen_service.rb:33]: Lint/UselessAssignment: Useless assignment to variable - `adyen_error`. Use `_` or `_adyen_error` as a variable name to indicate that it won't be used. (https://rubystyle.guide#underscore-unused-vars)

**Pull request merged**